dynamo/runner.py

```diff
@@ -0,0 +1,353 @@
+        f_kwargs: Dict[str, Any],
+        out_t: str,
+        dtype: torch.dtype = torch.float32,
+        device: torch.device = torch.device('cpu'),
+):
+    def _unpack_arg(arg,  out_t=out_t, dtype=dtype, device=device):
+        if callable(arg):
+            return arg(out_t=out_t, dtype=dtype, device=device)
+        return arg
+
+    return {
+        k: _unpack_arg(v)
+        for k, v in f_kwargs.items()
+    }
+
+
+def run(
+        configs: List[Dict[str, Any]],
+        output_filename: str,
+        verbose: bool
+) -> int:
+    with open(output_filename, 'wb') as fp:
+        for operator, input_type, device, optimize in _iter_op_device():
+            _opt_name, _opt_txt, _opt = optimize
+            _op_dev_txt = (f'\033[1;33m {operator} at {device} {_opt_txt}'
+                           '\033[0;0m')
+            print(
+                '-'*79,
+                f'\n-> Benchmarking{_op_dev_txt}'
+            )
+
+            for cfg, bs, res in _iter_cfg(configs):
+                x = create_inputs(
+                    bs, res, input_type,
+                    device=torch.device(device),
+                )
+
+                kwargs = _build_kwargs(
+                                       cfg['kwargs'],
+                                       out_t=input_type,
+                                       device=torch.device(device)
+                )
+
+                module_name = cfg['module']
+                import_from = f'{cfg["import_from"]}.{module_name}'
+
+                _args_values_str = ', '.join(
+                                    str(tuple(v.shape)) if hasattr(v, 'shape')
+                                    else str(v)
+                                    for v in kwargs.values()
+                )
+                sub_label = f'[{bs}, {res}, {_args_values_str}]'
+
+                print(
+                    '\n\n\t', '-'*70, '\n'
+                    f'\t->({_op_dev_txt}) Module: {module_name} |'
+                    f'Batch size={bs}, '
+                    f'resolution={res}, args={_args_values_str}',
+                )
+
+                if _check_run(
+                        verbose, import_from, operator, x, _opt, **kwargs
+                ):
+                    for num_threads in cfg['threads']:
+                        print(
+                            '\t\t-> benchmarking with '
+                            f'num_threads={num_threads}...',
+                        )
+
+                        desc = f'{_opt_name}{operator.split("_")[0]}_{device}'
+                        stmt = f'{operator}(input, **kwargs)'
+                        setup = f'from {import_from} import {operator}'
+
+                        bench_out = benchmark.Timer(
+                            stmt=stmt,
+                            setup=setup,
+                            globals={'input': x, 'kwargs': kwargs},
+                            num_threads=num_threads,
+                            label=module_name,
```

Comment:
TODO: `env=` and have an appropriate colect_env for our case  -- that can collect information from the system, that also works on hardware that we care about, such as a jetson

- https://github.com/rbonghi/jetson_stats/tree/master
- https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py

We need to collect the version/commit of which kornia was used too